packages/next/src/server/app-render/app-render.tsx

```diff
@@ -1426,8 +1426,7 @@ export const renderToHTMLOrFlight: AppPageRender = (
   query,
   renderOpts
 ) => {
-  // TODO: this includes query string, should it?
-  const pathname = validateURL(req.url)
+  const { pathname } = validateURL(req.url)
```

Comment:
Are we wanting to provide the search params separate? Currently we rely on this including query for populating `SearchParamsContext` here:

https://github.com/vercel/next.js/blob/29d53c87ea53336cf1e436119975d8e53bc28827/packages/next/src/client/components/router-reducer/create-initial-router-state.ts#L85-L91

https://github.com/vercel/next.js/blob/29d53c87ea53336cf1e436119975d8e53bc28827/packages/next/src/client/components/app-router.tsx#L331

https://github.com/vercel/next.js/blob/29d53c87ea53336cf1e436119975d8e53bc28827/packages/next/src/server/app-render/app-render.tsx#L452